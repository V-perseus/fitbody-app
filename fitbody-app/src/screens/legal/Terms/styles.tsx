import { StyleSheet } from 'react-native'
import globals from '../../../config/globals'

const styles = StyleSheet.create({
  container: {
    ...globals.styles.container,
    backgroundColor: globals.styles.colors.colorWhite,
  },
  view: {
    ...globals.styles.view,
  },
  headerTitle: {
    ...globals.header.headerTitleStyle,
    color: globals.styles.colors.colorBlack,
  },
  html: {
    flex: 1,
  },
})

export default styles
