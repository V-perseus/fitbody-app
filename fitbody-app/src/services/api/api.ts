import * as Sentry from '@sentry/react-native'
import deviceInfoModule from 'react-native-device-info'
import { Platform } from 'react-native'

// Services
import { setErrorMessage } from '../error'
import { displayLoadingModal, hideLoadingModal } from '../loading'
import NavigationService from '../NavigationService'
import globals from '../../config/globals'
import { updateUserIsRestricted } from '../../data/user'
import { storeRegistry } from '../../store/storeRegistry'

const API_BASE = globals.apiBase

type Methods = 'get' | 'post' | 'patch' | 'put' | 'delete'
type MethodsUppercase = 'GET' | 'POST' | 'PATCH' | 'PUT' | 'DELETE'
interface IDefaultHeaders {
  [key: string]: string
}
interface IParams {
  method: MethodsUppercase
  headers: IDefaultHeaders
  body?: FormData | string
}
export interface IAppErrorResponse {
  response: {
    data: {
      message: string
    }
  }
}
export function isAppErrorResponse(error: any): error is IAppErrorResponse {
  return error.response?.data?.message !== undefined
}

export const apiPublic = <T>(url: string, method: Methods = 'get', data: {} | null = {}, loader = true, headers = {}): Promise<T> => {
  // console.log('api',`${API_BASE}${url}`);
  if (loader) {
    displayLoadingModal()
  }

  let token = storeRegistry.getState().services.session.apiToken
  let defaultHeaders: IDefaultHeaders = {
    'Content-Type': 'application/json',
    Accept: 'application/json',
    'X-App-Version': deviceInfoModule.getVersion(),
    'X-App-Os': Platform.OS,
  }
  if (token) {
    defaultHeaders.Authorization = 'Bearer ' + token
  }

  let reqHeaders = {}
  Object.assign(reqHeaders, defaultHeaders)
  let params: IParams = {
    method: method.toUpperCase() as MethodsUppercase,
    headers: reqHeaders,
  }

  if ((method.toUpperCase() === 'POST' || method.toUpperCase() === 'PUT' || method.toUpperCase() === 'PATCH') && data) {
    if (data instanceof FormData) {
      params.body = data
    } else {
      params.body = JSON.stringify(data)
    }
  }

  let req = new Request(API_BASE + url, params)

  return fetch(req)
    .then((response) => {
      if (response.ok) {
        if (loader) {
          hideLoadingModal()
        }
        let responseText = response.text()
        return responseText
          .then(function (text) {
            return text ? Promise.resolve(JSON.parse(text)) : Promise.resolve({})
          })
          .catch(function (error) {
            return Promise.reject(error)
          })
      }
      if (loader) {
        hideLoadingModal()
      }
      return Promise.reject(response)
    })
    .catch((error: Response) => {
      let message: string | Error = ''
      if (error && error instanceof Error) {
        try {
          Sentry.captureException(new Error(error.message || JSON.stringify(error)))
        } catch {}
        message = error
      } else if (error && error.json) {
        error.json().then((errorJson) => {
          message = errorJson.message
          if (message && typeof message === 'string' && message.toLowerCase().includes('please purchase')) {
            updateUserIsRestricted(true)
            NavigationService.sendToPaywall()
          } else if (!url.includes('/v4/events')) {
            setErrorMessage({ error: message })
          }
        })
      }
      hideLoadingModal()
      return Promise.reject(message)
    })
}

/**
 * Sign a user in
 */
export const apiAuth = <T>(url: string, method: Methods = 'get', data: null | {} = {}, loader = true, headers = {}): Promise<T> => {
  return apiPublic<T>(url, method, data, loader, headers)
}
