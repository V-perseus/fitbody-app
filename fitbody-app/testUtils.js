import React from 'react'
import { render as rtlRender } from '@testing-library/react-native'
import { configureStore } from '@reduxjs/toolkit'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { Provider } from 'react-redux'

// Import your own reducer
import { appReducer } from './src/store'
import { ModalStackScreens } from './src/config/routes/loggedInStack'

// custom render method override
function render(ui, { preloadedState, store = configureStore({ reducer: appReducer, preloadedState }), ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react-native'
// override render method
export { render }

// custom navigation mock
const Stack = createStackNavigator()
export const MockedNavigator = ({
  component,
  params = {},
  preloadedState,
  store = configureStore({ reducer: appReducer, preloadedState }),
}) => {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="MockedScreen" component={component} initialParams={params} />
          <Stack.Screen name="Modals" component={ModalStackScreens} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  )
}
