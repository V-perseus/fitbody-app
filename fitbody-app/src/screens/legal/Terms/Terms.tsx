import React, { useEffect, useLayoutEffect, useState } from 'react'
import { SafeAreaView } from 'react-native-safe-area-context'
import { Text, View, StatusBar } from 'react-native'
import { WebView } from 'react-native-webview'
import { NavigationProp } from '@react-navigation/native'

// API
import api from '../../../services/api'

// Assets
import styles from './styles'

// Components
import { HeaderButton } from '../../../components/Buttons/HeaderButton'

// Types
import { AccountStackParamList } from '../../../config/routes/routeTypes'

interface ITermsProps {
  navigation: NavigationProp<AccountStackParamList, 'Terms'>
}
const Terms = ({ navigation }: ITermsProps) => {
  useLayoutEffect(() => {
    navigation.setOptions({
      headerTitle: () => <Text style={styles.headerTitle}>TERMS AND CONDITIONS</Text>,
      headerLeft: () => <HeaderButton onPress={() => navigation.goBack()} />,
    })
  }, [navigation])

  const [terms, setTerms] = useState('')
  // const [version, setVersion] = useState('')
  // const [updatedAt, setUpdatedAt] = useState('')

  useEffect(() => {
    api.legal.terms().then((data) => {
      setTerms(data.terms.content)
      // setVersion(data.terms.version)
      // setUpdatedAt(data.terms.updated_at)
    })
  }, [])

  return (
    <SafeAreaView style={styles.container}>
      <StatusBar barStyle="dark-content" />
      <View style={styles.view}>
        <WebView style={styles.html} testID="webview" originWhitelist={['*']} source={{ html: terms }} />
      </View>
    </SafeAreaView>
  )
}

export default Terms
