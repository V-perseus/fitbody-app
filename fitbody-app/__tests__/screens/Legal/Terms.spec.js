import api from '../../../src/services/api'
import Terms from '../../../src/screens/legal/Terms'
import { MockedNavigator, render, act, cleanup, waitFor } from '../../../testUtils'

jest.mock('../../../src/services/api')

describe('Terms Screen', () => {
  afterEach(() => {
    cleanup()
  })

  let comp = null
  beforeEach(async () => {
    comp = MockedNavigator({ component: Terms })
  })

  it('renders snapshot correctly', async () => {
    await act(async () => {
      await api.legal.terms.mockResolvedValue({ terms: { content: 'These are the Term And Conditions.' } })
      const tree = render(comp).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  it('Should render screen title', async () => {
    await act(async () => {
      const content = '<div>These are the Term And Conditions.</div>'
      await api.legal.terms.mockResolvedValue({ terms: { content } })
      const { queryByText } = render(comp)
      expect(queryByText(/Terms and Conditions/gi)).not.toBeNull()
    })
  })

  it('should render text in webview', async () => {
    const content = '<div>These are the Term And Conditions.</div>'
    await act(async () => {
      await api.legal.terms.mockResolvedValue({ terms: { content } })
      const { getByTestId } = render(comp)
      await waitFor(() => {
        expect(getByTestId('webview')).toHaveProp('source', { html: content })
      })
    })
  })
})
