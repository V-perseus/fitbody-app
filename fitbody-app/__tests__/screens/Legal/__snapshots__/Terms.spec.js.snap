// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Terms Screen renders snapshot correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  hitSlop={16}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "opacity": 1,
                      },
                      Object {
                        "alignItems": "center",
                        "borderRadius": 30,
                        "display": "flex",
                        "height": 42,
                        "justifyContent": "center",
                        "marginHorizontal": 7,
                        "width": 42,
                      },
                      undefined,
                    ]
                  }
                >
                  <arrow-left.svg
                    color="#333333"
                    testID="back_arrow"
                  />
                </View>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 590,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#333333",
                      "fontFamily": "OpenSans-Semibold",
                      "fontSize": 16,
                      "textAlign": "center",
                    }
                  }
                >
                  TERMS AND CONDITIONS
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RNCSafeAreaView
                        edges={
                          Array [
                            "bottom",
                            "left",
                            "right",
                            "top",
                          ]
                        }
                        style={
                          Object {
                            "backgroundColor": "#ffffff",
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "flexDirection": "column",
                              "marginHorizontal": 15,
                              "marginTop": 60,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                  "overflow": "hidden",
                                },
                                undefined,
                              ]
                            }
                          >
                            <RNCWebView
                              cacheEnabled={true}
                              injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
                              injectedJavaScriptForMainFrameOnly={true}
                              javaScriptEnabled={true}
                              messagingEnabled={false}
                              onContentProcessDidTerminate={[Function]}
                              onHttpError={[Function]}
                              onLoadingError={[Function]}
                              onLoadingFinish={[Function]}
                              onLoadingProgress={[Function]}
                              onLoadingStart={[Function]}
                              onMessage={[Function]}
                              onShouldStartLoadWithRequest={[Function]}
                              source={
                                Object {
                                  "html": "",
                                }
                              }
                              style={
                                Array [
                                  Object {
                                    "flex": 1,
                                    "overflow": "hidden",
                                  },
                                  Object {
                                    "backgroundColor": "#ffffff",
                                  },
                                  Object {
                                    "flex": 1,
                                  },
                                ]
                              }
                              testID="webview"
                              textInteractionEnabled={true}
                              useSharedProcessPool={true}
                            />
                          </View>
                        </View>
                      </RNCSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
